src/rules/attr.js

```diff
@@ -55,7 +59,10 @@ function executeOnElm($elm, config, reporter, ast) {
                     || conf instanceof Array
                     || typeof conf === "string"
                     || conf instanceof RegExp) {
-                if (attrs[attrib] === undefined) {
+                const optional = isAttrOptional(attrib);
+                const attr = optional ? attrib.substring(0, -1) : attrib;
+            
+                if (attrs[attr] === undefined && !optional) {
```

Comment:
This _seems_ unnecessarily complicated to me. Correct me if I'm wrong but I believe that if the attribute is optional, then this whole branch isn't relevant. [Like for `SPECIAL_ATTRIBS` we can exit early](https://github.com/birjj/svglint/blob/65f16bd6a1492456b67d1d617f4c0461eb3f45b5/src/rules/attr.js#L51).

If so, I suggest we rewrite this whole block as:

```js
// check that all attributes that must exist do so
Object.keys(config).forEach(
    attrib => {
        // do nothing with special configs
        if (SPECIAL_ATTRIBS.includes(attrib)) { return; }
        // do nothing for optional attributes
        if (isAttrOptional(attrib)) { return; }
        // if it must exist
        const conf = config[attrib];
        if (conf === true
                || conf instanceof Array
                || typeof conf === "string"
                || conf instanceof RegExp) {
            if (attrs[attrib] === undefined) {
                reporter.error(
                    `Expected attribute '${attrib}', didn't find it`,
                    $elm,
                    ast
                );
            }
        }
    }
);
```

Reply:
Hmmm, fair. I think the way it's currently written with nested conditionals is a little confusing, so maybe that's where I was getting tripped up. Do you know why there's the check there against the type of `config[attrib]`? Are there values allowed in the `attr` config that we *don't* want to use to check? It seems like it could be simplified to "if `config[attrib]` exists and `attrs[attrib]` doesn't, explode" if the type-checking was removed.

Reply:
You can see what the implications are if you remove the condition you mentioned from the test output (though unfortunately the tests don't actually fail :disappointed:), details below. Most notably configuring an attribute to `false` no longer works - note that it doesn't check if `conf` is a boolean, but rather if it's `true`.

Also, I _think_ (though I didn't write this piece of code) the if-statements are nested to create a separation in "what" is being checked. I.e., the outer if-statement runs checks against `conf` and the inner if-statement runs checks against `attrs`/`attrib`. If you don't mind, I'd just leave this as is for the purposes of this Pull Request.

---

Applying this diff on `main`:

```diff
      Object.keys(config).forEach(
          attrib => {
              // do nothing with special configs
              if (SPECIAL_ATTRIBS.includes(attrib)) { return; }
              // if it must exist
-             const conf = config[attrib];
-             if (conf === true
-                     || conf instanceof Array
-                     || typeof conf === "string"
-                     || conf instanceof RegExp) {
                  if (attrs[attrib] === undefined) {
                      reporter.error(
                          `Expected attribute '${attrib}', didn't find it`,
                          $elm,
                          ast
                      );
                  }
-             }
          }
      );
```

Results in the following (truncated) `npm test` output:

```log
...
Rule: attr
    ✔ should succeed without config
    ✔ should succeed with a required attribute
    ✔ should fail without a required attribute
    ✔ should succeed without a disallowed attribute
Error: Linting failed (error): { foo: false, 'rule::selector': 'svg' }
    at Linting.<anonymous> (file:///.../svglint/test/attr.spec.js:62:19)
    at Linting.emit (node:events:513:28)
    at Linting._onRuleFinish (file:///.../svglint/src/lib/linting.js:161:18)
    at file:///.../svglint/src/lib/linting.js:136:26
    at file:///.../svglint/src/lib/linting.js:110:33
...
```

Reply:
That wasn't quite the diff I was proposing, just pushed dc61069 and #3329b24 which is more like what I was proposing and does pass all the tests.

Reply:
Thanks for elaborating, this seems like a reasonable improvement :+1: